.circleci/config.yml

```diff
@@ -0,0 +1,86 @@
+version: 2
+
+main: &main
+    machine: true
+    environment:
+      # Used by qthelpers to close widgets after a defined time
+      - TEST_CI: "True"
+      - TEST_CI_APP: "True"
+    steps:
+      - checkout
+      - run:
+          command: docker pull dorowu/ubuntu-desktop-lxde-vnc:trusty
+      - run:
+          name: Install system packages
+          command: |
+            sudo apt-get update
+            sudo apt-get install libegl1-mesa
+      - run:
+          name: Install Miniconda
+          command: |
+            wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+            bash miniconda.sh -b -p $HOME/miniconda
+            export PATH="$HOME/miniconda/bin:$PATH"
+            conda install python=$PYTHON_VERSION
+      - run:
+          name: Install Spyder from GitHub
+          command: |
+            export PATH="$HOME/miniconda/bin:$PATH"
+            conda install --only-deps spyder
+            mkdir spyder-source
+            cd spyder-source
+            wget -q https://github.com/spyder-ide/spyder/archive/3.x.zip
+            unzip -q 3.x.zip
+            cd spyder-3.x
+            python setup.py install
+      - run:
+          name: Set up test environment
+          command: |
+            export PATH="$HOME/miniconda/bin:$PATH"
+            conda install -q notebook pytest pytest-cov flaky
+            pip install coveralls pytest-qt
+            python setup.py develop
+            echo '********** output of conda list **********'
+            conda list
+      - run:
+          name: Run tests
+          command: |
+            export PATH="$HOME/miniconda/bin:$PATH"
+            mkdir test-reports
+            pytest spyder_notebook --cov=spyder_notebook --junitxml=test-reports/junit.xml
```

Comment:
This command needs to be

    pytest -x -vv spyder_notebook --cov=spyder_notebook

Reply:
And after this command please add these lines

```bash
if [ $? -ne 0 ]; then
    exit 1
fi
```

so that this step fails if the `pytest`  exit code is not 0.